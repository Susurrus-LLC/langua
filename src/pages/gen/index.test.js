import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { PureGen as Gen } from '.'

describe('Gen', () => {
  const data = {
    site: {
      siteMetadata: {
        toolInfo: {
          gen: {
            title: 'Gen',
            link: 'gen',
            description: 'Gen description'
          }
        }
      }
    }
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Gen data={data} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    let tree = renderer.create(<Gen data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
