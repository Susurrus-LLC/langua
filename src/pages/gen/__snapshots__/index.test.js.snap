// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gen renders accurately 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <div
      className="titleArea"
    >
      <h1
        className="siteTitle"
      >
        <a
          className="titleLink"
          href="/"
        />
      </h1>
      <p
        className="subtitle"
      />
    </div>
    <nav
      className="nav"
    >
      <ul
        className="menu"
      >
        <li
          className="navLink"
        >
          <a
            href="/gen"
          >
            Gen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/morph"
          >
            Morph
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/deriv"
          >
            Deriv
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/frequen"
          >
            Frequen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/tree"
          >
            Tree
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/lex"
          >
            Lex
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    <a
      className="btn buttonLink"
      href="/gen/help"
    >
      Help
    </a>
    <h2
      className="toolTitle"
    >
      undefinedGen
    </h2>
    <form
      className="form"
      name="gen-form"
    >
      <div
        className="subpatterns part"
      >
        <h5
          className="sectionTitle"
        >
          Subpatterns
        </h5>
        <div
          className="subpsRow"
        >
          <select
            aria-label="Variable name for subpattern 0"
            className="select"
            id="v0"
            name="v0"
            onChange={[Function]}
            value="V"
          >
            <option
              value="A"
            >
              A
            </option>
            <option
              value="B"
            >
              B
            </option>
            <option
              value="C"
            >
              C
            </option>
            <option
              value="D"
            >
              D
            </option>
            <option
              value="E"
            >
              E
            </option>
            <option
              value="F"
            >
              F
            </option>
            <option
              value="G"
            >
              G
            </option>
            <option
              value="H"
            >
              H
            </option>
            <option
              value="I"
            >
              I
            </option>
            <option
              value="J"
            >
              J
            </option>
            <option
              value="K"
            >
              K
            </option>
            <option
              value="L"
            >
              L
            </option>
            <option
              value="M"
            >
              M
            </option>
            <option
              value="N"
            >
              N
            </option>
            <option
              value="O"
            >
              O
            </option>
            <option
              value="P"
            >
              P
            </option>
            <option
              value="Q"
            >
              Q
            </option>
            <option
              value="R"
            >
              R
            </option>
            <option
              value="S"
            >
              S
            </option>
            <option
              value="T"
            >
              T
            </option>
            <option
              value="U"
            >
              U
            </option>
            <option
              value="V"
            >
              V
            </option>
            <option
              value="W"
            >
              W
            </option>
            <option
              value="X"
            >
              X
            </option>
            <option
              value="Y"
            >
              Y
            </option>
            <option
              value="Z"
            >
              Z
            </option>
          </select>
          <input
            aria-label="Subpattern 0"
            className="subpatternInput"
            id="p0"
            name="p0"
            onChange={[Function]}
            type="text"
            value="a/e/i/o/u"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="c0"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="subpsRow"
        >
          <select
            aria-label="Variable name for subpattern 1"
            className="select"
            id="v1"
            name="v1"
            onChange={[Function]}
            value="C"
          >
            <option
              value="A"
            >
              A
            </option>
            <option
              value="B"
            >
              B
            </option>
            <option
              value="C"
            >
              C
            </option>
            <option
              value="D"
            >
              D
            </option>
            <option
              value="E"
            >
              E
            </option>
            <option
              value="F"
            >
              F
            </option>
            <option
              value="G"
            >
              G
            </option>
            <option
              value="H"
            >
              H
            </option>
            <option
              value="I"
            >
              I
            </option>
            <option
              value="J"
            >
              J
            </option>
            <option
              value="K"
            >
              K
            </option>
            <option
              value="L"
            >
              L
            </option>
            <option
              value="M"
            >
              M
            </option>
            <option
              value="N"
            >
              N
            </option>
            <option
              value="O"
            >
              O
            </option>
            <option
              value="P"
            >
              P
            </option>
            <option
              value="Q"
            >
              Q
            </option>
            <option
              value="R"
            >
              R
            </option>
            <option
              value="S"
            >
              S
            </option>
            <option
              value="T"
            >
              T
            </option>
            <option
              value="U"
            >
              U
            </option>
            <option
              value="V"
            >
              V
            </option>
            <option
              value="W"
            >
              W
            </option>
            <option
              value="X"
            >
              X
            </option>
            <option
              value="Y"
            >
              Y
            </option>
            <option
              value="Z"
            >
              Z
            </option>
          </select>
          <input
            aria-label="Subpattern 1"
            className="subpatternInput"
            id="p1"
            name="p1"
            onChange={[Function]}
            type="text"
            value="p/t/k/b/d/g/s/m/n/l/r"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="c1"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="subpsRow"
        >
          <select
            aria-label="Variable name for subpattern 2"
            className="select"
            id="v2"
            name="v2"
            onChange={[Function]}
            value="N"
          >
            <option
              value="A"
            >
              A
            </option>
            <option
              value="B"
            >
              B
            </option>
            <option
              value="C"
            >
              C
            </option>
            <option
              value="D"
            >
              D
            </option>
            <option
              value="E"
            >
              E
            </option>
            <option
              value="F"
            >
              F
            </option>
            <option
              value="G"
            >
              G
            </option>
            <option
              value="H"
            >
              H
            </option>
            <option
              value="I"
            >
              I
            </option>
            <option
              value="J"
            >
              J
            </option>
            <option
              value="K"
            >
              K
            </option>
            <option
              value="L"
            >
              L
            </option>
            <option
              value="M"
            >
              M
            </option>
            <option
              value="N"
            >
              N
            </option>
            <option
              value="O"
            >
              O
            </option>
            <option
              value="P"
            >
              P
            </option>
            <option
              value="Q"
            >
              Q
            </option>
            <option
              value="R"
            >
              R
            </option>
            <option
              value="S"
            >
              S
            </option>
            <option
              value="T"
            >
              T
            </option>
            <option
              value="U"
            >
              U
            </option>
            <option
              value="V"
            >
              V
            </option>
            <option
              value="W"
            >
              W
            </option>
            <option
              value="X"
            >
              X
            </option>
            <option
              value="Y"
            >
              Y
            </option>
            <option
              value="Z"
            >
              Z
            </option>
          </select>
          <input
            aria-label="Subpattern 2"
            className="subpatternInput"
            id="p2"
            name="p2"
            onChange={[Function]}
            type="text"
            value="m/n"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="c2"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="addingRow"
        >
          <button
            className="btn success small"
            id="add"
            onClick={[Function]}
            role="button"
            type="button"
          >
            Add
          </button>
        </div>
      </div>
      <div
        className="pattern part"
      >
        <h5
          className="sectionTitle"
        >
          Pattern
        </h5>
        <input
          aria-label="Pattern"
          className="patternInput"
          id="pattern"
          name="pattern"
          onChange={[Function]}
          type="text"
          value="V/CV/CVN/CVCV/CVCVN"
        />
      </div>
      <div
        className="control part"
      >
        <div
          className="controlSide controlLeft"
        >
          <div
            className="controlPiece"
          >
            <button
              className="btn neutral"
              id="generate"
              onClick={[Function]}
              role="button"
              type="submit"
            >
              Generate
            </button>
          </div>
          <div
            className="controlPiece"
          >
            <label
              htmlFor="words"
            >
              words:
            </label>
            <input
              aria-label="Number of words to generate"
              className="wordsInput"
              id="words"
              max="9999"
              min="1"
              name="words"
              onChange={[Function]}
              type="number"
              value={100}
            />
          </div>
          <div
            className="controlPiece"
          >
            <label>
              <input
                checked={false}
                id="newline"
                name="options"
                onChange={[Function]}
                type="checkbox"
                value="newline"
              />
               
              new line each
            </label>
          </div>
          <div
            className="controlPiece"
          >
            <label>
              <input
                checked={false}
                id="filterdupes"
                name="options"
                onChange={[Function]}
                type="checkbox"
                value="filterdupes"
              />
               
              filter duplicates
            </label>
          </div>
        </div>
        <div
          className="controlSide controlRight"
        >
          <div
            className="controlPiece"
          >
            <button
              className="btn success"
              id="save"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Save
            </button>
          </div>
          <div
            className="controlPiece"
          >
            <button
              className="btn"
              id="open"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Open
            </button>
            <input
              className="hidden"
              id="file"
              name="file"
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </div>
    </form>
    <div
      className="results"
    >
      <div
        className="output"
      >
        <p
          className="outText"
        >
          
        </p>
      </div>
      <div
        className="stats"
      >
        <p
          className="statsText"
        >
          words: 0; maximum different words: 0
        </p>
      </div>
    </div>
    <div
      className="Toastify"
    />
  </main>
  <footer
    className="footer"
  >
    <p
      className="copyright"
    >
      <a
        href="undefined/blob/master/CHANGELOG.md"
        rel="noopener noreferrer"
        target="_blank"
      >
        Version 
        <i
          className="fas fa-code"
        />
         
      </a>
      . Built by
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      />
      , copyright © 
      2017–2020
       under the
       
      <a
        href="https://choosealicense.com/licenses/agpl-3.0/"
        rel="noopener noreferrer"
        target="_blank"
      >
        AGPL-3.0 
        <i
          className="fas fa-gavel"
        />
         license
      </a>
      .
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        View this project on 
        <i
          className="fab fa-github"
        />
         GitHub.
      </a>
    </p>
  </footer>
</div>
`;
